addons/sale/models/sale.py

```diff
@@ -182,7 +182,7 @@ def _search_invoice_ids(self, operator, value):
 
     user_id = fields.Many2one(
         'res.users', string='Salesperson', index=True, tracking=2, default=lambda self: self.env.user,
-        domain=lambda self: [('groups_id', 'in', self.env.ref('sales_team.group_sale_salesman').id)])
+        domain=lambda self: [('groups_id', 'in', self.env.ref('sales_team.group_sale_salesman').id), ('share', '=', False), ('company_id', '=', self.env.company.id)])
```

Comment:
Not your fault, but I've some problems in mind here :D
If the company of the SO is different than the main one in the environment, it'll only allow users from the env company.

I fear that the only correct solution would be a dynamical domain `domain="[('is_salesman', '=', True), ('company_ids', 'in', [company_id])]"`, along with a new non stored computed field on res.users, with a `search` attribute & method able to convert `[('is_salesman', '=', True)]` to `[('groups_id', 'in', self.env.ref('sales_team.group_sale_salesman').id)]`

That way, we would have a correct multi company filter AND a correct filter on salesman only (note that the `[('share', '=', False)]` part is implied by the salesman group, which is only given to internal users anyway).

Reply:
Hey, I'm not sure to understand exactly how to write this method. Should the method just handle = True or = False ? Also this change would need to be in master as I'm creating a new field right ?

Reply:
> I'm not sure to understand exactly how to write this method
I don't know if you ever had to work with the search method of a field, but you can find a lot of them in the code.  They receive the operator & value, and are expected to return a domain :).  I believe supporting True & False should be enough for the case where it'll be used :ok_hand:.

You can add a field in stable as long as it is not stored AFAIK.

Reply:
if I understand, the new field is only so we can combine:

-  `('company_ids', 'in', [company_id])` that needs to be a string domain,
- and `('groups_id', new_operator, self.env.ref('sales_team.group_sale_salesman').id)` that needs to be a lambda domain?

an alternative might be to return a string in a lambda domain:

```py
          domain=lambda self: "[('groups_id', 'in', {}), ('share', '=', False), ('company_ids', '=', [company_id])]".format(self.env.ref('sales_team.group_sale_salesman').id)
```

Reply:
It's working, I just had to change `company_ids` => `company_id`